cfme/tests/openstack/cloud/test_flavors.py

```diff
@@ -199,3 +236,45 @@ def test_flavor_details(appliance, soft_assert, private_flavor):
 
     soft_assert(view.entities.relationships.get_text_of('Cloud Provider') ==
                 private_flavor.provider.name)
+
+
+@pytest.mark.rfe
+@pytest.mark.ignore_stream('5.9')
+def test_create_instance_with_private_flavor(instance_with_private_flavor, provider, soft_assert):
+    view = navigate_to(instance_with_private_flavor, 'Details')
+    prov_data = provider.data['provisioning']
+    power_state = view.entities.summary('Power Management').get_text_of('Power State')
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = view.entities.summary('Relationships').get_text_of('VM Template')
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    flavors = [f.name for f in provider.mgmt.api.flavors.list()]
+    flavor = view.entities.summary('Relationships').get_text_of('Flavor')
+    soft_assert(flavor in flavors)
+
+    # Assert other relationships in a loop
+    props = [('Availability Zone', 'availability_zone'),
+             ('Cloud Tenants', 'cloud_tenant'),
+             ('Virtual Private Cloud', 'cloud_network')]
+
+    for p in props:
+        v = view.entities.summary('Relationships').get_text_of(p[0])
+        soft_assert(v == prov_data[p[1]])
+
+
+    collection = provider.appliance.collections.cloud_flavors
+    flavor_obj = collection.instantiate(flavor, provider)
+    soft_assert(flavor_obj.instance_count == 1)
+
+    flavor_obj.delete()
+    view = provider.appliance.browser.create_view(navigator.get_class(collection, 'All').VIEW)
```

Comment:
Required:
In such case please just use
`view = naviagate_to(collection, "All")`
If you are on the same page. then it will automatically avoid redundant navigation; Actually @jadhaj that's beauty of `navimazing`.